src/addon-manager.js

```diff
@@ -383,10 +383,16 @@ class AddonManager extends EventEmitter {
                  typeof(templateVal) + ', found: ' + typeof(objectVal);
         }
         if (typeof(objectVal) === 'object') {
-          let err = this.validateManifestObject(prefix + key + '.',
-                                                objectVal, templateVal);
-          if (err) {
-            return err;
+          if (Array.isArray(objectVal)) {
+            if (templateVal.length > 0 && objectVal.length === 0) {
+              return `Expecting ${prefix}${key} to be a non-empty array.`;
+            }
```

Comment:
Should this also validate the type of the array entries? Currently, it would accept files: [1, 2, 3]
We can probably make the simplifying assumption that each element of the array should be of the same type.
We probably don't need to support empty arrays.

Reply:
Sure, we can do that.